madara/crates/client/db/src/lib.rs

```diff
@@ -593,12 +612,14 @@ impl MadaraBackend {
         chain_config: Arc<ChainConfig>,
         config: MadaraBackendConfig,
     ) -> anyhow::Result<Self> {
+        let chain_head = ChainHead::load_from_db(&db).context("Getting latest block_n from database")?;
```

Comment:
this chain_head is already taken from db inside of the `open` function.
IMO - you should put you set `starting_block` inside of that function too and not here, for consistency.
You can just default `starting_block` to be None in this constructor, and replace the value in the `open` function